### PR TITLE
Update Akka.TestKit.NUnit to NUnit v4 (3rd attempt)

### LINKED ISSUE #111: Update TestKit to use NUnit v4
⚠️ NOTE: This change would entail dropping `netstandard2.0` support for `Akka.TestKit.NUnit` and moving to `net6.0`, because this is is now a constraint of NUnit itself.

**Is your feature request related to a problem? Please describe.**

NUnit v4 was released, which has some breaking changes. We'd like to utilize the latest NUnit and keep it updated.

**Describe the solution you'd like**

Per conversation with Aaron, we think the best approach so far is to have Akka.TestKit.NUnit use the latest version of NUnit, which is now v4.

**Describe alternatives you've considered**

We considered a separate package, but there's already an Akka.TestKit.NUnit3 package which should be sufficient for these purposes.

**Additional context**

NUnit v4 has some breaking changes around assertions etc.

### PR DESCRIPTION
**Pull request by milang**
Fixes #111.

## Changes

This PR extends pull request #131 by @UrsMetz, which in turn builds on the work of @SeanKilleen in #112.

### Improvements in NUnit TestKit adapters

- ⚠️ `Akka.TestKit.NUnit`
    - Uses the latest NUnit 4 (`4.2.2`)
    - Targets `net6.0` and `net462`, just like NUnit 4
- ⚠️ `Akka.TestKit.NUnit3`
    -  Uses the latest NUnit 3 (`3.14.0`)
    - Targets `netstandard2.0`, just like NUnit 3
        - NUnit 3 actually supports `net35;net40;net45;netstandard2.0`, but we have to exclude frameworks not supported by `Akka.TestKit@1.5.28`
- Implementations of NUnit 4 and NUnit 3 adapters are now almost identical, both using NUnit's constraint model. The only difference is how the assertion message is constructed, because NUnit 4 removed support for [format and params specification](https://docs.nunit.org/articles/nunit/release-notes/breaking-changes.html#nunit-40:~:text=Assert.That%20overloads%20with%20format%20specification%20and%20params%20have%20been%20removed%20in%20favor%20of%20an%20overload%20using%20FormattableString).
- `ITestKitAssertions.AssertEqual(..., comparer, ...)` now uses NUnit constraint model instead of custom assertion implementation. This makes failure reports more consistent with the rest of NUnit output.
- All dependencies have been bumped to the latest version. Merging this PR should allow all existing dependabot pull requests to be closed (except `Nuke.Common`).
    dotnet-outdated

### Improvements in NuGet packages

- Icon and license are now specified by `PackageIcon` and `PackageLicenseExpression` properties, eliminating warnings during `dotnet pack`. Akka icon with dimensions 128x128px is now embedded in `.nupkg` file, as [recommended by NuGet](https://github.com/NuGet/Home/wiki/Packaging-Icon-within-the-nupkg#:~:text=Size%20recommendation%20%2D%20128x128).
- Symbol files (`.pdb`) are now included directly in the `.nupkg` files. This only increases the final `.nupkg` size by ~20 kB and removes the need to publish separate `.snupkg` file to a symbol server. See [Microsoft guidance](https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/nuget#symbol-packages).
- Current year is automatically set in copyright message

The following screenshot compares currently published `.nupkg` file (version `1.5.24`) with `.nupkg` as proposed by this pull request. Please note that the green "source link", "deterministic exe/dll", and "compiler flags" shields typically require the package to be built with the latest .NET SDK (mine was built using .NET SDK 8.0.401). However this repository's GitHub Actions are currently configured to use .NET SDK 6.

Akka.TestKit.NUnit.nupkg comparison

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,10 +1,20 @@
+## [1.6.0] / September 2024
+
+- Bump `Akka.TestKit` to [1.5.27.1](https://github.com/akkadotnet/akka.net/releases/tag/1.5.27.1)
+- Bump `NUnit3TestAdapter` to [4.6.0](https://github.com/nunit/nunit3-vs-adapter/releases/tag/V4.6.0)
+- Add `NUnit.Analyzers` version [4.3.0](https://github.com/nunit/nunit.analyzers/releases/tag/4.3.0) to ensure we follow NUnit best practices.
+- Add support for [NUnit 4.2.2](https://github.com/nunit/nunit/releases/tag/4.2.2) in addition to NUnit 3.
+   `Akka.TestKit.NUnit` targets NUnit 4, while `Akka.TestKit.NUnit` targets NUnit 3.
```
Comment (FarrisPandell) [inline RELEASE_NOTES.md]:
```suggestion
   `Akka.TestKit.NUnit` targets NUnit 4, while `Akka.TestKit.NUnit3` targets NUnit 3.
```

**Review by FarrisPandell — CHANGES_REQUESTED** [review]

File: RELEASE_NOTES.md (hunk 1 above)
Comment (milang) [reply RELEASE_NOTES.md]: Updated.

**Comment by milang** [comment]
@FarrisPandell addressed your feedback, ready for next round.

**Review by FarrisPandell — APPROVED** [review]

**Comment by milang** [comment]
@Aaronontheweb any feedback for this pull request? I updated the `Akka.TestKit` dependency to `1.5.28`, new `Akka.TestKit.NUnit` release would nicely match yesterday's [Akka.NET release](https://github.com/akkadotnet/akka.net/releases/tag/1.5.28)?

**Comment by Aaronontheweb** [comment]
@milang @FarrisPandell my apologies - I've had GitHub Notifications turned off for 10 years (unless I get explicitly mentioned), so this is the first I'm seeing this. CI/CD is failing - any idea what that's about?

**Comment by Aaronontheweb** [comment]
Overall the NuGet package structure you've proposed looks good to me

**Comment by milang** [comment]
@Aaronontheweb 

> CI/CD is failing - any idea what that's about?

Nuke is complaining that _"Release Notes should not be empty"_. I tracked it down to newlines that I added after version headers in `RELEASE_NOTES.md` - the newlines were recommended by `markdownlint`, but they were confusing Nuke build.

I pushed a new commit that removes the empty newlines, hopefully the build will pass now.

**Comment by Aaronontheweb** [comment]
> @Aaronontheweb
> 
> > CI/CD is failing - any idea what that's about?
> 
> Nuke is complaining that _"Release Notes should not be empty"_. I tracked it down to newlines that I added after version headers in `RELEASE_NOTES.md` - the newlines were recommended by `markdownlint`, but they were confusing Nuke build.
> 
> I pushed a new commit that removes the empty newlines, hopefully the build will pass now.

Ugh, yes NUKE is very opinionated about this. Ran into that myself before.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
So it's release as v1.6.0-preview1 on NuGet - let me know if that's acceptable

**Comment by Aaronontheweb** [comment]
I'm also fine doing it as v1.5.* package if there's buy-in

**Comment by UrsMetz** [comment]
@Aaronontheweb would a release as 1.6.* mean that a corresponding Akka 1.6.* was needed to use the NUnit v4 compatible version of Akka.TestKit.NUnit?
I think currently all Akka packages tend to share the same minor version.
My team is eagerly waiting for the NUnit 4 compatibility.

**Comment by SeanKilleen** [comment]
@milang thanks for getting this across the finish line! Awesome work 🎉 👏

**Comment by milang** [comment]
@Aaronontheweb 

> So it's release as v1.6.0-preview1 on NuGet - let me know if that's acceptable
> I'm also fine doing it as v1.5.* package if there's buy-in

@UrsMetz 

> Would a release as 1.6.* mean that a corresponding Akka 1.6.* was needed to use the NUnit v4 compatible version of Akka.TestKit.NUnit? I think currently all Akka packages tend to share the same minor version.

Sorry, I didn't realize that the major/minor version was fixed at `1.5` to match rest of Akka.NET. Initially I bumped to `1.6` to follow semantic versioning, indicating that major version of NUnit dependency changed. That being said, I have no issue with staying on `1.5`, and in light of Urs' comment that would be the better approach.

@Aaronontheweb don't forget to delete the [`VersionSuffix` line #7 in `Directory.Build.props`](https://github.com/akkadotnet/Akka.TestKit.NUnit/blob/dev/src/Directory.Build.props#L7) to remove the pre-release flag.

---

Edit: I see that the `dev` branch was tagged as `1.5.28` after this PR was merged. I think it makes a lot of sense to publish this build as `1.5.28` to NuGet.

**Comment by milang** [comment]
@Aaronontheweb any chance you could publish `Akka.TestKit.NUnit` as version `1.5.28` (see my comments from last week) ?

Also, I noticed that `Akka.TestKit.NUnit3` has not been published in a while, and is at version `1.3.8`:
image

As per the changes in this PR, the latest build of `Akka.TestKit.NUnit3` should be published as version `1.5.28`. It can remain obsolete, but IMO the deprecation message should be updated from:

> This version is now deprecated - please use Akka.TestKit.NUnit (which also supports NUnit 3)

to:

> This version is now deprecated - please use Akka.TestKit.NUnit with NUnit 4.

**Pull request merged**